pkg/controllers/disruption/controller.go

```diff
@@ -213,7 +214,7 @@ func (c *Controller) executeCommand(ctx context.Context, m Method, cmd Command,
 	// tainted with the Karpenter taint, the provisioning controller will continue
 	// to do scheduling simulations and nominate the pods on the candidate nodes until
 	// the node is cleaned up.
-	schedulingResults.Record(logging.WithLogger(ctx, operatorlogging.NopLogger), c.recorder, c.cluster)
+	schedulingResults.Record(log.IntoContext(ctx, zapr.NewLogger(zap.NewNop())), c.recorder, c.cluster)
```

Comment:
Rather than re-creating this logger every time, can we create it once and like we do with the `operatorlogging.NopLogger` and then just ref out to it when we need it? This is going to avoid us having to continually create the empty logger here

Reply:
Done.